engine/src/main/java/org/terasology/world/block/entity/BlockEntitySystem.java

```diff
@@ -163,8 +163,9 @@ private void commonDestroyed(DoDestroyEvent event, EntityRef entity, Block block
         if (blockDamageModifierComponent == null || !blockDamageModifierComponent.skipPerBlockEffects) {
             // dust particle effect
             if (entity.hasComponent(LocationComponent.class) && block.isDebrisOnDestroy()) {
+                //TODO: particle system stuff should be split out better - this is effectively a stealth dependency on
```

Comment:
This was a TODO even before this PR... @kBlaszczyk you just touched the particle system ... 🙄 Jokes aside, how can we get this out of the engine?